File: py/1028.py
def gcd(a, b):
    if b == 0:
        return a
    else:
        return gcd(b, a%b)

n = int(input())
count = 0
while count < n:
    deck1, deck2 = map(int, input().split())
    print(gcd(deck1, deck2))

    count += 1

File: cpp/1028.cpp
#include <iostream>

using namespace std;

int gcd(int a, int b)
{
    if (b == 0) {
        return a;
    }

    return gcd(b, a % b);
}

int main()
{
    int n, a, b;
    cin >> n;

    for (n; n > 0; n--) {
        cin >> a >> b;
        cout << gcd(a, b) << '\n';
    }

    return 0;
}